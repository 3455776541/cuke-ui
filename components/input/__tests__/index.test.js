import React from "react";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Input from "../index";

describe("<Input/>", () => {
  it("should render Input", () => {
    const wrapper = render(
      <div>
        <Input type="text" />
        <Input type="password" placeholder="请输入" />
        <Input type="number" placeholder="请输入" />
        <Input readonly value="我是只读" />
        <Input disabled placeholder="禁用" />
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render addon with input", () => {
    const wrapper = render(
      <div>
        <Input addonBefore={"www"} placeholder="请输入" />
        <Input addonAfter={".com"} placeholder="填写网址" />
        <Input
          addonBefore={"https://"}
          addonAfter={".cn"}
          placeholder="www.lijinke"
        />
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should emit onChange events", () => {
    const onChange = jest.fn();
    const wrapper = shallow(<Input onChange={onChange} type="text" />);

    wrapper.find(".cuke-input").simulate("change", {
      persist: () => {},
      target: {
        value: null
      }
    });
    expect(onChange).toHaveBeenCalled();
  });

  it("should render custom size", () => {
    const wrapper = render(
      <div>
        <Input placeholder="small" size="small" />
        <Input placeholder="default" />
        <Input placeholder="large" size="large" />
      </div>
    );
    expect(wrapper.find(".cuke-input-small").length === 1);
    expect(wrapper.find(".cuke-input-large").length === 1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should set new state when props value update", () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <Input onChange={onChange} type="text" value={1} />
    );

    expect(wrapper.state().value).toBe(1);
    wrapper.setProps({
      value: 3
    });
    expect(wrapper.state().value).toBe(3);
  });
});
