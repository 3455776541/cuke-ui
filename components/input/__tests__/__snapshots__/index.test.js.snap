// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> should render Input 1`] = `
<div>
  <input
    class="cuke-input"
    placeholder=""
    type="text"
  />
  <input
    class="cuke-input"
    placeholder="请输入"
    type="password"
  />
  <input
    class="cuke-input"
    placeholder="请输入"
    type="number"
  />
  <input
    class="cuke-input"
    placeholder=""
    readonly=""
    type="text"
    value="我是只读"
  />
  <input
    class="cuke-input cuke-input-disabled"
    disabled=""
    placeholder="禁用"
    type="text"
  />
</div>
`;

exports[`<Input/> should render addon with input 1`] = `
<div>
  <span
    class="cuke-input-group cuke-input-group-addon-before"
  >
    <span
      class="cuke-input-group-addon"
    >
      www
    </span>
    <input
      class="cuke-input"
      placeholder="请输入"
      type="text"
    />
  </span>
  <span
    class="cuke-input-group cuke-input-group-addon-after"
  >
    <input
      class="cuke-input"
      placeholder="填写网址"
      type="text"
    />
    <span
      class="cuke-input-group-addon"
    >
      .com
    </span>
  </span>
  <span
    class="cuke-input-group cuke-input-group-addon-before cuke-input-group-addon-after cuke-input-group-addon-all"
  >
    <span
      class="cuke-input-group-addon"
    >
      https://
    </span>
    <input
      class="cuke-input"
      placeholder="www.lijinke"
      type="text"
    />
    <span
      class="cuke-input-group-addon"
    >
      .cn
    </span>
  </span>
</div>
`;

exports[`<Input/> should render custom size 1`] = `
<div>
  <input
    class="cuke-input cuke-input-small"
    placeholder="small"
    type="text"
  />
  <input
    class="cuke-input"
    placeholder="default"
    type="text"
  />
  <input
    class="cuke-input cuke-input-large"
    placeholder="large"
    type="text"
  />
</div>
`;
